File: back/src/main/java/com/shop_monitor/back/Model/Tick.java
package com.shop_monitor.back.Model;

import com.fasterxml.jackson.annotation.JsonFormat;
import jakarta.persistence.*;
import lombok.Data;
import lombok.RequiredArgsConstructor;

import java.time.LocalDateTime;

@Data
@Entity
@RequiredArgsConstructor
public class Tick {
    @Id
    @GeneratedValue(strategy= GenerationType.AUTO)
    Long id;
    boolean open;
    @JsonFormat(pattern="yyyy-MM-dd'T'HH:mm:ss.SSS")
    @Column(name="dateTime")
    LocalDateTime dateTime;


    public Tick(boolean b, LocalDateTime now) {
        this.open = b;
        this.dateTime = now;
    }

    public String toString(){
        return "date: " + String.valueOf(this.dateTime) + "- Bool: " + String.valueOf(this.open);
    }
}

File: mobile_app/app/src/main/java/com/z_ha/shopmonitor/model/Tick.kt
package com.z_ha.shopmonitor.model

import java.time.LocalDate
import java.time.LocalDateTime
import java.util.Date

class Tick {
    val open : String?
    val dateTime: String?
    val id: Long = 0


    constructor(status: String, date: String){
        this.open = status
        this.dateTime = date
    }


    override fun toString(): String {
        return "isOpen: ${this.open}, dateTime: ${this.dateTime}"
    }

}